<launch>

  <arg name="camera" default="/fisheye_cam/image"/>
  <arg name="calib" default="calib_results.txt"/>
  <arg name="bagfile" default="performance_test.bag"/>

  <param name="/use_sim_time" value="true"/>
  <node pkg="rosbag" type="play" name="rosbag" args="--clock $(arg bagfile)"/>

  <!-- Republishes the compressed topic so it can be used to image_proc node -->
  <node name="republish" type="republish" pkg="image_transport" output="screen" args="compressed in:=$(arg camera) raw out:=$(arg camera)" />

  <node pkg="tf" type="static_transform_publisher" name="odom_to_map" args="0 0 0 0 -1.57079632679 -1.57079632679 visual_odom map 30"/>

  <!-- Run the viso2_ros package -->
  <node pkg="viso2_ros" type="mono_odometer_omnidirectional" name="mono_odometer" output="screen">
      <param name="image" value="$(arg camera)"/>

      <param name="max_features" value="10" />
      <param name="motion_threshold" value="100" />
      <param name="camera_height" value="1.12" />
      <param name="camera_pitch"  value="0.00" />

      <param name="base_link_frame_id" value="base_link"/>
      <param name="publish_tf" value="true"/>
      <param name="odom_frame_id" value="visual_odom"/>

      <param name="calib_path" value="$(arg calib)" />
  </node>

</launch>
